compiler/rustc_const_eval/src/transform/validate.rs

```diff
@@ -546,6 +546,29 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
 
 impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
     fn visit_operand(&mut self, operand: &Operand<'tcx>, location: Location) {
+        match operand {
```

Comment:
this code should be removed? 🤔 I would expect to instead change all `sub` in the validator except in `ProjectionElem::Subtype` to be changed to `eq` once were in the runtime MIR phase